tests/network/primary_pod_network.go

```diff
@@ -74,16 +78,69 @@ var _ = SIGDescribe("Primary Pod Network", func() {
 			var vmi *v1.VirtualMachineInstance
 
 			BeforeEach(func() {
-				vmi = vmlifecycle.SetupVMI(vmiWithMasqueradeBinding())
+				vmi = libvmi.SetupVMI(vmiWithMasqueradeBinding())
 			})
 
 			AfterEach(func() {
-				vmlifecycle.CleanupVMI(vmi)
+				libvmi.CleanupVMI(vmi)
 			})
 
 			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
 		})
 	})
+
+	Describe("Connectivity", func() {
+		const servicePort = 1500
+
+		AssertConnectivity := func(outboundVMI, inboundVMI *v1.VirtualMachineInstance) {
+			inboundVMIAddress := inboundVMI.Status.Interfaces[0].IP
+
+			By("Checking connectivity via ICMP")
+			Expect(tests.PingFromVMConsole(outboundVMI, inboundVMIAddress)).To(Succeed())
+
+			By("Checking connectivity via ICMP with MTU-sized frames")
+			mtu := getPodNetworkMTU(outboundVMI)
```

Comment:
e2e tests are expensive and I'm not convinced that having this negative test would bring enough value. I'm not strongly against it, but I'd need to see a strong motivation. Do you suspect this would be something that could fail. Or would the fail signal a regression in our code?

Reply:
well I always try to add at lease 1 negative scenario, for the remote chance that the test **always** passes regardless of the input, but that's just my paranoid self :) 
I agree that it's not crucial here and if overall test duration is a considerable issue to consider here then maybe it's best to leave it.